codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -140,6 +143,100 @@ public void onPriorityTreeParentChanging(Http2Stream stream, Http2Stream newPare
                 }
             }
         });
+
+        if (listener == null) {
```

Comment:
Taking another step back I wonder if it might be cleaner to use a strategy for writing, rather than using raw visitors throughout the code and if-then-else's for whether we have a listener or not.  Maybe something like:

```java
private interface class StreamWriter {
    boolean isWritable(Http2Stream stream);
    void incrementWindowSize(Http2Stream stream, int delta);
    void writeAllocatedBytes();
}
```

Then we could have 2 implementations: with and without a listener.  WDYT?